ucsb-cs156-f22/f22-5pm-courses
# glt - Adjust `DELETE` method for  `/api/courses/user` to delete primaries/secondaries together

## Overview

Update:  Tested on Heroku QA at f22-5pm-2-courses.

This PR should be merged after #21. Closes #18.

This PR updates the endpoint `DELETE /api/courses/user` to delete all related lectures and discussion sections for an argument `PSCourse` id.

To accomplish this, we first obtain the enroll code of the class, check the UCSB API for related sections, and then find all records in the database matching each enroll code (and in the same personal schedule) and delete each them.

## Tests

Test details are briefly explained in the [second commit](https://github.com/ucsb-cs156-f22/f22-5pm-courses/pull/36/commits/c469cfb8ff87dc63799d134f47dcafcdaf5dbd88) of this PR.

## How to test

1. Clone my branch.
2. Build and run.
3. Login and create a personal schedule. Remember your psId.
4. Add some classes:
  - A class with a primary only.
  - Two primary/secondary class pairs. (That is, four classes total.)
  - Remember the ids of the `PSCourse`s you add to your schedule!
5. Go to Swagger and under `ps-course-controller`, find the endpoint `DELETE /api/courses/user`.

![image](https://user-images.githubusercontent.com/72234957/204076293-84a3cb74-71a3-4445-a98f-327038b4febe.png)
6. Delete each class/class-pair from your schedule.
  - For the class that consists of only a primary, you should get a message like this:

![image](https://user-images.githubusercontent.com/72234957/204076581-04fd3165-6e0a-4b03-8c54-f5a2ce105002.png)

  - Delete a primary/secondary pair using the primary's `PSCourse` id:
  
![image](https://user-images.githubusercontent.com/72234957/204076428-ac9f32af-9077-47ec-83dd-4fd941a72ac6.png)

  - Delete the other primary/secondary pair using the secondary's `PSCourse` id:
  
![image](https://user-images.githubusercontent.com/72234957/204076596-e7acc48f-bef8-45e5-8d2f-b936068dfc7a.png)
